### PR TITLE
LF-3465 Delete a crop plan | Core functionality

### PR DESCRIPTION
**Pull request by kathyavini**
**Description**

This PR adds Crop Plan Deletion to the frontend and backend.

Backend notes:
- Following the pattern of abandon management plan, the records in the `management_plan` and `task` tables are identified and soft deleted, with the associations between task and plan defined in `management_tasks` are hard-deleted

Frontend notes:
- I have made the significant change of moving the Redux store updates on GET requests for tasks and management plans from `upsertMany` (which adds but does not remove entities) to `setAll` which resets the store for these slices in favour of the current database state
- This fixes many of the concurrency issues we saw before with tasks, and has been generally stable in my testing, but there may be more downstream effects we should keep an eye on going into sanity/release next week

Jira link: https://lite-farm.atlassian.net/browse/LF-3465

I have folded the concurrency issues into this PR so this is now also the branch for [LF-3467](https://lite-farm.atlassian.net/browse/LF-3467).

**Type of change**

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

**How Has This Been Tested?**

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [ ] Passes test case
- [x] UI components visually reviewed on desktop view
- [x] UI components visually reviewed on mobile view
- [ ] Other (please explain)

**Checklist:**

- [x] I have commented my code, particularly in hard-to-understand areas
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [x] The precommit and linting ran successfully
- [ ] I have added or updated language tags for text that's part of the UI
- [x] I have added "MISSING" for all new language tags to languages I don't speak
- [ ] I have added [the GNU General Public License](https://lite-farm.atlassian.net/l/cp/BT0Dd7WW) to all new files

[LF-3467]: https://lite-farm.atlassian.net/browse/LF-3467?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: packages/api/src/controllers/managementPlanController.js [hunk 1]
```diff
@@ -206,11 +206,97 @@ const managementPlanController = {
...
-        if (isDeleted) {
+          .where({ management_plan_id })
+          .where('deleted', false)
+          .first();
+
+        if (!managementPlan) {
+          return res.status(404).send('Management plan not found');
+        }
+
+        const result = await ManagementPlanModel.transaction(async (trx) => {
```
Comment (Duncan-Brain) [inline packages/api/src/controllers/managementPlanController.js:221]: I think this transaction object should be passed around: https://vincit.github.io/objection.js/guide/transactions.html#passing-around-a-transaction-object

Comment (kathyavini) [reply packages/api/src/controllers/managementPlanController.js:221]: I took this verbatim from the existing abandonment controller so I assumed it was correct 🙂 I'll take a close read through that documentation -- looks at first glance like both patterns are permissible.

Comment (kathyavini) [reply packages/api/src/controllers/managementPlanController.js:310]: Okay, read it and added 😄 Thank you!

Do you think the abandonment should be updated now too? Or a separate ticket?

**Review by SayakaOno — APPROVED** [review]
Great job on figuring things out! It works great 👏

File: packages/api/src/controllers/managementPlanController.js (hunk 1 above)
Comment (Duncan-Brain) [reply packages/api/src/controllers/managementPlanController.js:310]: definitely separate ticket lol

**Review by Duncan-Brain — APPROVED** [review]
Worked great for me, that query looked tough!

**Comment by Duncan-Brain** [comment]
@kathyavini Actually looks like we got merge conflicts here

File: packages/webapp/src/containers/managementPlanSlice.js [hunk 2]
```diff
@@ -44,7 +44,7 @@ const addManyManagementPlan = (state, { payload: managementPlans }) => {
   state.loading = false;
   state.error = null;
   state.loaded = true;
-  managementPlanAdapter.upsertMany(
+  managementPlanAdapter.setAll(
```
[the remark this answers is not in the document]
Comment (iperdomo) [reply packages/webapp/src/containers/managementPlanSlice.js:47]: As commented by @kathyavini this change from `upsertMany` to `setAll` is to take _deletes_ into account.
https://redux-toolkit.js.org/api/createEntityAdapter#crud-functions

**Review by Duncan-Brain — DISMISSED** [review]
Works for me! Great find with the setall

**Review by Duncan-Brain — APPROVED** [review]

File: packages/api/src/controllers/managementPlanController.js (hunk 1 above)
Comment (iperdomo) [reply packages/api/src/controllers/managementPlanController.js:310]: I would like to understand the status of this comments. In the same controller we have 2 different ways of handling a transaction commit/rollback. Ideally we use one pattern. I'm not sure if just passing `trx` ensures commit at the end?

Comment (kathyavini) [reply packages/api/src/controllers/managementPlanController.js:336]: Yes, passing `trx` is enough for commit or rollback (it's pattern #2 in the documentation @Duncan-Brain linked above and the pattern in the primary example). We have plenty of both patterns in the codebase. I did a quick search and found an [older Jira ticket](https://lite-farm.atlassian.net/browse/LF-355) suggesting we should move towards the passing `trx` pattern, but we have a complete mix, so not sure if getting them aligned is at all in the scope of this ticket.

**Review by iperdomo — APPROVED** [review]
I'll merge the changes. Any issue should be addressed in a separate ticket.

**Pull request merged**